sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/OffHeapColumnVector.java

```diff
@@ -85,6 +85,7 @@ public long nullsNativeAddress() {
 
   @Override
   public void close() {
+    super.close();
     Platform.freeMemory(nulls);
```

Comment:
This comment is not related to this fix directly. However, is it better to check whether each field is not `0` before calling `Platform.freeMemory()`? For example, `data` or `lengthData/offsetData may be `0`.

Reply:
Yeah, it is a bit weird. The doc in the JDK8's `Unsafe. freeMemory()` states the following: `The address passed to this method may be null, in which case no action is taken.` See: http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/sun/misc/Unsafe.java